drivers/adc/adc_mcux_lpadc.c

```diff
@@ -88,37 +91,88 @@ static int mcux_lpadc_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
+	/* Select ADC CMD register to configure based off channel ID */
+	cmd = &data->cmd_config[channel_cfg->channel_id];
+
+	/* If bit 5 of input_positive is set, then channel side B is used */
+	channel_side = 0x20 & channel_cfg->input_positive;
+	/* Channel number is selected by lower 4 bits of input_positive */
+	channel_num = ADC_CMDL_ADCH(channel_cfg->input_positive);
+
+	LOG_DBG("Channel num: %u, channel side: %c", channel_num,
+		channel_side == 0 ? 'A' : 'B');
+
+	LPADC_GetDefaultConvCommandConfig(cmd);
+
 	if (channel_cfg->differential) {
-		LOG_ERR("Differential channels are not supported");
+		/* Channel pairs must match in differential mode */
+		if ((channel_cfg->input_positive | 0x20) !=
```

Comment:
doesn't this conditional mean that input_negative can't be an A channel? also this doesn't seem to error check the case where the same B channel is specified twice

Reply:
Yes- the driver rewrite used a KConfig to determine if the channel calculation would be A-B or B-A. However, there isn't any reason this has to be done with Kconfig, so I've moved that setting into the driver (so now, setting input_negative as an A channel will result in the calculation being A-B)

I don't really see any value in a runtime check to make sure both channels aren't the same though. If a user puts the same channel for `input-negative` and `input-positive`, they shouldn't expect defined behavior, since they are trying to take the difference between two identical channels. We could error check this, but I don't see why we'd always verify the user hadn't made this mistake, since nothing about the ADC API suggests this would be supported behavior